mood-analysis.rb

```diff
@@ -19,6 +20,55 @@ def analyze_mood(words)
   return ":-|"
 end
 
+def get_sanitized_input(words)
+  # input now matches feelings hash 
+  words.downcase!
+  # did not write a strip_punctuation method but this line strips it
+  words.gsub!(/[^a-z0-9\s]/i, '')
+  return words
+end 
+
+def get_happy_days(emoji_array)
+  happy = 0 
+  smile = ":-)"
+
+  number_of_days = 1 
+  emoji_array.each do |emoji|
+    if emoji == smile
+      happy += 1
+      if happy >= 3 && happy < 4
```

Comment:
I suspect you did this because it was printing the following line multiple times. Once for the first time it found 3 happy days and then for every day after that. You could eliminate that by fixing the condition as you did, but then the loop continues to iterate when we don't need it to. Instead you could have had a `return` that gets you out of the method instead of continuing to loop.